### PR TITLE
Feature field access

### PR DESCRIPTION
**Pull request by flbulgarelli**
Fixes https://github.com/mumuki/mulang/issues/203

Introducing `FieldReference` and `FieldAssignment` expressions, and adding support for Java, Python and JavaScript.

### CODE REVIEW

**Comment by flbulgarelli** [comment]
@fedescarpa If you want just add C support for fields to this branch

File: src/Language/Mulang/Inspector/Generic.hs [hunk 1]
```diff
@@ -69,12 +69,21 @@ assignsMatching matcher predicate = containsExpression f
...
+        f _                              = False
+
 -- | Inspection that tells whether an expression uses the the given target identifier
 -- in its definition
 uses :: BoundInspection
 uses p = containsExpression f
   where f = any p . referencedIdentifiers
 
+usesField :: BoundInspection
+usesField p = containsExpression f
```
Comment (julian-berbel) [inline src/Language/Mulang/Inspector/Generic.hs:84]: I have two concerns regarding this:
- Should the definition be the exact same as `uses`'? What if I just use a variable with the same name as a field, and never access the field itself?
- We don't seem to be exporting this in the module (`assignsFieldMatching` either, for that matter)

Comment (flbulgarelli) [reply src/Language/Mulang/Inspector/Generic.hs:84]: All this function look like I just copy pasted the `uses` inspection. It should call `assignsFieldMatching` instead.

**Comment by flbulgarelli** [comment]
Related to https://github.com/mumuki/mulang/issues/163

File: src/Language/Mulang/Inspector/Generic.hs (hunk 1 above)
Comment (flbulgarelli) [reply src/Language/Mulang/Inspector/Generic.hs]: In order to continue blocking this issue, I am moving the new inspections to another branch.

**Pull request merged**